base/methodshow.jl

```diff
@@ -1,5 +1,11 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+function MethodTable(m::Method)
+    ft = ccall(:jl_first_argument_datatype, Any, (Any,), m.sig)
+    ft == C_NULL && error("Method ", m, " does not correspond to a function type")
```

Comment:
The function will return a null reference, not `C_NULL`, so this check doesn't work. Fortunately that shouldn't happen for any type that comes from a `Method.sig` field.

I also don't much like this being a constructor of `MethodTable`, since it doesn't construct one.

Reply:
Fair enough. I could go back to calling it `get_methodtable`.